lib/__tests__/factory-builders.test.ts

```diff
@@ -226,6 +231,38 @@ describe('params', () => {
     expect(adminFactory.build().admin).toBe(true);
     expect(adminFactory.build().admin).toBe(true);
   });
+
+  it('adds parameters for a sub-type which are then also accepted in build()', () => {
+    const adminFactory = userFactory.params<AdminUser>({
+      admin: true,
```

Comment:
My concern is that if `admin: true` were not supplied, then the factory happily compiles but does not actually return an `AdminUser`.